app/build.gradle.kts

```diff
@@ -0,0 +1,172 @@
+    return process.inputStream.bufferedReader().use { it.readText().trim() }
+}
+
+val gitCommitHash = fetchGitCommitHash()
+val keyProps = Properties()
+val keyPropsFile: File = rootProject.file("signature/keystore.properties")
+if (keyPropsFile.exists()) {
+    println("Loading keystore properties from ${keyPropsFile.absolutePath}")
+    keyProps.load(FileInputStream(keyPropsFile))
+}
+
+android {
+    compileSdk = 34
+
+    defaultConfig {
+        applicationId = "me.ash.reader"
+        minSdk = 26
+        targetSdk = 33
+        versionCode = 25
+        versionName = "0.10.1"
+
+        buildConfigField("String", "USER_AGENT_STRING", "\"ReadYou/${'$'}{versionName}(${versionCode})\"")
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+        vectorDrawables.useSupportLibrary = true
+
+        room {
+            schemaDirectory("$projectDir/schemas")
+        }
+
+        ksp {
+            arg("room.incremental","true")
+        }
+    }
+
+    flavorDimensions.add("channel")
+    productFlavors {
+        create("github") {
+            dimension = "channel"
+        }
+        create("fdroid") {
+            dimension = "channel"
+        }
+        create("googlePlay") {
+            dimension = "channel"
+            applicationIdSuffix = ".google.play"
+        }
+    }
+    signingConfigs {
+        create("release") {
+            keyAlias = keyProps["keyAlias"] as String?
+            keyPassword = keyProps["keyPassword"] as String?
+            storeFile = keyProps["storeFile"]?.let { file(it as String) }
+            storePassword = keyProps["storePassword"] as String?
+        }
+    }
+    lint {
+        disable.addAll(listOf("MissingTranslation", "ExtraTranslation"))
+    }
+    buildTypes {
+        getByName("release") {
+            isMinifyEnabled = true
+            isShrinkResources = true
+            proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro")
+            signingConfig = signingConfigs.getByName("release")
+        }
+    }
+    applicationVariants.all {
+        outputs.all {
+            (this as com.android.build.gradle.internal.api.BaseVariantOutputImpl).outputFileName = "ReadYou-${defaultConfig.versionName}-${gitCommitHash}.apk"
+        }
+    }
+    kotlinOptions {
+        freeCompilerArgs = freeCompilerArgs + "-opt-in=kotlin.RequiresOptIn"
+        jvmTarget = JavaVersion.VERSION_11.toString()
+    }
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
+    }
+    buildFeatures {
+        compose = true
+        buildConfig = true
+    }
+    composeOptions {
+        kotlinCompilerExtensionVersion = libs.versions.composeCompiler.get()
+    }
+    packaging {
+        resources.excludes.add("/META-INF/{AL2.0,LGPL2.1}")
+    }
+    androidResources {
+        generateLocaleConfig = true
+    }
+    namespace = "me.ash.reader"
+}
+
+dependencies {
```

Comment:
we have divided these dependencies into several groups before (separated with empty lines), preserve such structure could make this section look not that teriffying for newcomers